module_clean_urls.lua

```diff
@@ -0,0 +1,737 @@
+    "btsid",
+    "ws_ab_test",
+    "*@amazon.*",
+    "callback@bilibili.com",
+    "cvid@bing.com",
+    "form@bing.com",
+    "sk@bing.com",
+    "sp@bing.com",
+    "sc@bing.com",
+    "qs@bing.com",
+    "pq@bing.com",
+    "sc_cid",
+    "mkt_tok",
+    "trk",
+    "trkCampaign",
+    "ga_*",
+    "gclid",
+    "gclsrc",
+    "hmb_campaign",
+    "hmb_medium",
+    "hmb_source",
+    "spReportId",
+    "spJobID",
+    "spUserID",
+    "spMailingID",
+    "itm_*",
+    "s_cid",
+    "elqTrackId",
+    "elqTrack",
+    "assetType",
+    "assetId",
+    "recipientId",
+    "campaignId",
+    "siteId",
+    "mc_cid",
+    "mc_eid",
+    "pk_*",
+    "sc_campaign",
+    "sc_channel",
+    "sc_content",
+    "sc_medium",
+    "sc_outcome",
+    "sc_geo",
+    "sc_country",
+    "nr_email_referer",
+    "vero_conv",
+    "vero_id",
+    "yclid",
+    "_openstat",
+    "mbid",
+    "cmpid",
+    "cid",
+    "c_id",
+    "campaign_id",
+    "Campaign",
+    "hash@ebay.*",
+    "fb_action_ids",
+    "fb_action_types",
+    "fb_ref",
+    "fb_source",
+    "fbclid",
+    "refsrc@facebook.com",
+    "hrc@facebook.com",
+    "gs_l",
+    "gs_lcp@google.*",
+    "ved@google.*",
+    "ei@google.*",
+    "sei@google.*",
+    "gws_rd@google.*",
+    "gs_gbg@google.*",
+    "gs_mss@google.*",
+    "gs_rn@google.*",
+    "_hsenc",
+    "_hsmi",
+    "__hssc",
+    "__hstc",
+    "hsCtaTracking",
+    "source@sourceforge.net",
+    "position@sourceforge.net",
+    "t@*.twitter.com",
+    "s@*.twitter.com",
+    "ref_*@*.twitter.com",
+    "t@*.x.com",
+    "s@*.x.com",
+    "ref_*@*.x.com",
+    "t@*.fixupx.com",
+    "s@*.fixupx.com",
+    "ref_*@*.fixupx.com",
+    "t@*.fxtwitter.com",
+    "s@*.fxtwitter.com",
+    "ref_*@*.fxtwitter.com",
+    "t@*.twittpr.com",
+    "s@*.twittpr.com",
+    "ref_*@*.twittpr.com",
+    "t@*.fixvx.com",
+    "s@*.fixvx.com",
+    "ref_*@*.fixvx.com",
+    "tt_medium",
+    "tt_content",
+    "lr@yandex.*",
+    "redircnt@yandex.*",
+    "feature@*.youtube.com",
+    "kw@*.youtube.com",
+    "si@*.youtube.com",
+    "pp@*.youtube.com",
+    "si@*.youtu.be",
+    "wt_zmc",
+    "utm_source",
+    "utm_content",
+    "utm_medium",
+    "utm_campaign",
+    "utm_term",
+    "si@open.spotify.com",
+    "igshid",
+    "igsh",
+    "share_id@reddit.com",
+}
+
+local fixServices = {
+    ["bsky.app"] = "bskyx.app",
+    ["deviantart.com"] = "fxdeviantart.com",
+    ["instagram.com"] = "ddinstagram.com",
+    ["pixiv.net"] = "ppxiv.net",
+    ["reddit.com"] = "rxddit.com",
+    -- Currently broken
+    -- ["threads.net"] = "vxthreads.net",
+    ["tiktok.com"] = "tiktxk.com",
+    ["tumblr.com"] = "tpmblr.com",
+    ["twitch.tv"] = "fxtwitch.tv",
+    -- Use vxtwitter instead of fxtwitter since it includes greedy analytics
+    ["twitter.com"] = "vxtwitter.com",
+    ["x.com"] = "fixvx.com",
+}
+
+Module.UsersHanging = {}
+
+Module.UniversalRules = {}
+Module.HostRules = {}
+Module.RulesByHost = {}
+Module.GuildUniversalRules = {}
```

Comment:
Why not use guild persistent data?

Reply:
How does it work?

Reply:
You can use `self:GetPersistentData(guild)` to get a permanent (saved) per-guild table, and `self:GetGetData(guild)` to get temporary (wiped on bot reboot, ideal for caching) per-guild table.

Reply:
Sweet, ty, will implement this tomorrow